### ISSUE TITLE
`poetry update --only` does not work properly

### ISSUE THREAD

**Issue by Vuizur** [python-poetry/poetry#6452]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Windows 10
- **Poetry version**: ^1.2.0
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: [here](https://github.com/Vuizur/add-stress-to-epub/blob/c3a21b3c8f864ef825243269b845d5d3ecfee8b4/pyproject.toml)

## Issue

I wanted to update the optional Git dependency `ruwiktionary-html-parser`. To do this, I executed `poetry update --only "database-creation" to only the database-creation dependencies (where the dependency I wanted to update is included).
 
Now, when I run `poetry show -v` I get the displayed package version as `ruwiktionary-htmldump-parser 0.1.4 c649ece`. However, when I follow the link to the venv and look at the installed packages, I find the file `ruwiktionary_htmldump_parser-0.1.2.dist-info` and notice that the code is also not up to date. The update command only seems to have change the displayed version, but did not actually install anything.

**Comment by finswimmer** [comment]
Hello @Vuizur,

the `--only` parameter expects a `group` name. According to your `pyproject.toml` `database-creation` is an extra and not a [group](https://python-poetry.org/docs/managing-dependencies/#dependency-groups). IIRC we don't have a shortcut to update dependencies listed as `extra`. You have to update them separat, e.g. `poetry update ruwiktionary-htmldump-parser`.

fin swimmer

**Comment by Vuizur** [comment]
Thank you very much for the answer. Oups, I confused that. However, the problem seems to have remained. I tried to run `poetry update ruwiktionary-htmldump-parser`, it changed the version as shown by poetry show, but did not install anything.

The only way that works for me is running `poetry install -E "GUI"` and then `poetry install -E "GUI database-creation"`, which uninstalls all packages in the database-creation extra and then reinstalls them. This is unfortunately pretty slow.

**Comment by dimbleby** [comment]
Trying to make sense of what is reoprted here, it sounds like 

- `poetry update package` is successfully updating the lockfile, 
- but if `package` is an optional extra then it does not install the updated version

So you have to run `poetry install -E extra` afterwards.

**Comment by Vuizur** [comment]
> Trying to make sense of what is reoprted here, it sounds like
> 
> * `poetry update package` is successfully updating the lockfile,
> * but if `package` is an optional extra then it does not install the updated version

Exactly. My package is an optional extra and links to a github repo. I don't know if the same problem applies to PyPi packages. Updating non-extra packages works fine (I only tested the ones from PyPi though). 

I also get no message like "Updating package XXX" like usual. It seems like only the lockfile (?), or whatever `poetry show` reads, gets updated.

> So you have to run `poetry install -E extra` afterwards.
Yes, this seems to be the easiest way to actually update the package.

**Comment by Vuizur** [comment]
Correction: `poetry install -E extra` doesn't seem to work, at least in my case where I have to list two extras, so uninstalling and reinstalling is the only option.

**Comment by danyeaw** [comment]
I can reproduce this in poetry 1.3.2
